dotnet/samples/KernelSyntaxExamples/Example51_StepwisePlanner.cs

```diff
@@ -12,77 +15,213 @@
+            plannerConfig.MaxTokens = MaxTokens.Value;
         }
 
-        if (result.Variables.TryGetValue("skillCount", out string? skillCount))
+        SKContext result;
+        sw.Start();
+
+        try
+        {
+            StepwisePlanner planner = new(kernel: kernel, config: plannerConfig);
+            var plan = planner.CreatePlan(question);
+
+            result = await plan.InvokeAsync(kernel.CreateNewContext());
+
+            if (result.Result.Contains("Result not found, review _stepsTaken to see what", StringComparison.OrdinalIgnoreCase))
+            {
+                Console.WriteLine("Could not answer question in " + plannerConfig.MaxIterations + " iterations");
+                currentExecutionResult.answer = "Could not answer question in " + plannerConfig.MaxIterations + " iterations";
+            }
+            else
+            {
+                Console.WriteLine("Result: " + result.Result);
+                currentExecutionResult.answer = result.Result;
+            }
+
+            if (result.Variables.TryGetValue("stepCount", out string? stepCount))
+            {
+                Console.WriteLine("Steps Taken: " + stepCount);
+                currentExecutionResult.stepsTaken = stepCount;
+            }
+
+            if (result.Variables.TryGetValue("skillCount", out string? skillCount))
+            {
+                Console.WriteLine("Skills Used: " + skillCount);
+            }
+
+            if (result.Variables.TryGetValue("iterations", out string? iterations))
+            {
+                Console.WriteLine("Iterations: " + iterations);
+                currentExecutionResult.iterations = iterations;
+            }
+        }
+#pragma warning disable CA1031
+        catch (Exception ex)
         {
-            Console.WriteLine("Skills Used: " + skillCount);
+            Console.WriteLine("Exception: " + ex);
         }
 
         Console.WriteLine("Time Taken: " + sw.Elapsed);
+        currentExecutionResult.timeTaken = sw.Elapsed.ToString();
+        ExecutionResults.Add(currentExecutionResult);
         Console.WriteLine("*****************************************************");
     }
 
-    private static IKernel GetKernel()
+    private static IKernel GetKernel(ref ExecutionResult result, bool useChat = false, string? model = null)
     {
         var builder = new KernelBuilder();
+        if (useChat)
+        {
+            builder.WithAzureChatCompletionService(
+                model ?? ChatModelOverride ?? TestConfiguration.AzureOpenAI.ChatDeploymentName,
+                TestConfiguration.AzureOpenAI.Endpoint,
+                TestConfiguration.AzureOpenAI.ApiKey,
+                alsoAsTextCompletion: true,
+                setAsDefault: true);
+
+            Console.WriteLine($"Model: {model ?? ChatModelOverride ?? TestConfiguration.AzureOpenAI.ChatDeploymentName} ({ChatMaxTokens ?? (new Microsoft.SemanticKernel.Planning.Stepwise.StepwisePlannerConfig()).MaxTokens})");
```

Comment:
Improve formatting for better readability. It's important since example code should be easy to follow.